### PR TITLE
Implemented automatic semester term population rather than hardcoding

### PR DESCRIPTION
**Pull request by jackschedel**
PS: might be easier to review per commit rather than the entire "Files Changed" since the first commit is just formatting (plus deletion of one weird duplicate term index in UF.js)

### CODE REVIEW

**Comment by jackschedel** [comment]
Hold off on approving - I just realized a bug while in bed lol

File: server/src/client/Calendar.js [hunk 1]
```diff
@@ -50,116 +55,146 @@ class App extends React.Component {
...
   generatePeriods(props) {
     //Generate everything
-    let calendarData = this.props.calendar;//needs to be changed
+    let calendarData = this.props.calendar; //needs to be changed
+    const semesterTerm = this.props.semesterVal;
     const rows = periods.map((period, periodKey) => {
-      if (periodTimesSummer["semesters"].includes(this.props.semesterVal) && periodTimesSummer[period] || periodTimesNormal["semesters"].includes(this.props.semesterVal) && periodTimesNormal[period]) {
+      if (
+        (semesterTerm.charAt(3) === "5" && periodTimesSummer[period]) ||
+        (semesterTerm.charAt(3) !== "5" && periodTimesNormal[period])
```
Comment (Luc-Olsthoorn) [inline server/src/client/Calendar.js:81]: is there a way to keep this logic on the BE? it doesn't really matter, but its cleaner to keep it in config

Comment (jackschedel) [reply server/src/client/Calendar.js]: done. all changes are in services/UF.js now.

**Review by Luc-Olsthoorn — APPROVED** [review]
hope it works :)

**Comment by jackschedel** [comment]
It was just announced that Spring 2024 SOC opens on Sept. 29. Currently the implementation is set to iterate the Spring semester year on October 1st.

While we could technically change it to roll over on this specific day, there seems to be very little consistency between years. Advance registration for Spring doesn't even open until October 23, so the **two day lag period** seems acceptable to allow less hardcoding and better futureproofing.

Here's to hoping they keep the semester schema the same and my code works without modifications.

**Pull request merged**